vernac/vernacinterp.ml

```diff
@@ -31,22 +31,14 @@ let interp_typed_vernac (Vernacextend.TypedVernac { inprog; outprog; inproof; ou
     | Some stack, Update -> Some (LStack.map_top ~f:(fun _ -> proof') stack)
     | stack, New -> Some (LStack.push stack proof')
   in
+  let () =
+    match outproof with
```

Comment:
@maximedenes I'm unsure this is 100% correct, if you have a nested proof with a different mode, is the original mode restored when the inner proof is closed?

So we may need a stack for this.

Reply:
We should just have a test case, I guess.

Reply:
That'd be great, you can use `Load` to create such test case.